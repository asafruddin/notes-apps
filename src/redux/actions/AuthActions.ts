import { LOGIN } from "../../constant/ActionTypes"

export const StoreKey = (biometricKey: String) => {    
    return ({
        type: LOGIN,
        key: biometricKey,
    })
}