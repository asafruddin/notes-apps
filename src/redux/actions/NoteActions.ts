import { ADD_NOTES, UPDATE_NOTES } from "../../constant/ActionTypes";

export const AddNotes = (title: String, text: String, id: number) => {
    
    return ({
        type: ADD_NOTES,
        title: title,
        text: text,
        id: id
    });
}

export const UpdateNote = (title: String, text: String, id: number) => {
    return ({
        type: UPDATE_NOTES,
        title: title, 
        text: text,
        id: id
    })
}