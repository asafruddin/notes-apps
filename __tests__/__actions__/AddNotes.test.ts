import {expect, describe, it} from '@jest/globals'
import { AddNotes } from '../../src/redux/actions/NoteActions';
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from '../../src/redux/reducers/RootReducer';

describe('testing the add notes action', () => {
    const store = configureStore({
        reducer: rootReducer
    })

    it('user add note with title and notes succeed', async () => {
        await store.dispatch(AddNotes('Hello', 'This is the notes for hello title', 10));
        expect(store.getState).toMatchSnapshot()
    })
})