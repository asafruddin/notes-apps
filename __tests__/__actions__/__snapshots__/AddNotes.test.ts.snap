// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the add notes action user add note with title and notes succeed 1`] = `[Function]`;
